libfreerdp/core/peer.c

```diff
@@ -1234,6 +1234,8 @@ static void freerdp_peer_disconnect(freerdp_peer* client)
 
 	transport = freerdp_get_transport(client->context);
 	transport_disconnect(transport);
+	// the socket is now closed, don't attempt to close it again in freerdp_peer_free
+	client->sockfd = -1;
```

Comment:
but then the file descriptor would be leaked if the peer is freed without being disconnected. is this acceptable? the functions don't document what is guaranteed in this regard.

Reply:
@ggirol-rc well, it will not leak unless the whole `rdpTransport` leaks. (note: `transport_free` ensures `transport_disconnect` is called)
as for documentation, yes, could need a hand or two ;)